src/Parser/Shaman/Restoration/Modules/Talents/NaturesGuardian.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+
+import CooldownThroughputTracker from '../Features/CooldownThroughputTracker';
+/**
+ * When your health is brought below 35%, you instantly heal for 20% of your maximum health.
+ * Cannot occur more than once every 45 sec.
+ * 
+ * As this talent has no cast event, but you should still be able to see it on the timeline,
+ * this module creates that event whenever this talent heals the player.
+ */
+class NaturesGuardian extends Analyzer {
+  static dependencies = {
+    cooldownThroughputTracker: CooldownThroughputTracker,
+  };
+  healing = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.NATURES_GUARDIAN_TALENT.id);
+  }
+
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+
+    if (spellId !== SPELLS.NATURES_GUARDIAN_HEAL.id) {
+      return;
+    }
+
+    const castEvent = {...event};
+    castEvent.ability.guid = SPELLS.NATURES_GUARDIAN_TALENT.id;
```

Comment:
This mutates the original event! While you duplicated the event, the ability object will continue to reference to the same object.

Objects like this could be written like this to prevent such issues:

```jsx
this.owner.fabricateEvent(castEvent, {
  ...event,
  type: 'cast',
  ability: {
    ...event.ability,
    guid: SPELLS.NATURES_GUARDIAN_TALENT.id,
  },
});
```

Reply:
Oh, guess i misunderstood how {...event} works. Fixed